File: app/models/academy_video.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.sql import func
from sqlalchemy import DateTime

from app.database import Base


class AcademyVideo(Base):
    __tablename__ = "academy_videos"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    section = Column(String, nullable=False)
    youtube_id = Column(String, nullable=False)
    description = Column(String, nullable=True)
    sort_order = Column(Integer, default=0)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: mobile/lib/models/academy_video.dart
class AcademyVideo {
  final int id;
  final String title;
  final String section;
  final String youtubeId;
  final String? description;
  final int sortOrder;

  const AcademyVideo({
    required this.id,
    required this.title,
    required this.section,
    required this.youtubeId,
    this.description,
    required this.sortOrder,
  });

  factory AcademyVideo.fromJson(Map<String, dynamic> json) {
    return AcademyVideo(
      id: json['id'],
      title: json['title'],
      section: json['section'],
      youtubeId: json['youtube_id'],
      description: json['description'],
      sortOrder: json['sort_order'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'title': title,
        'section': section,
        'youtube_id': youtubeId,
        'description': description,
        'sort_order': sortOrder,
      };
}
